apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/model/ProjectTypeContainer.java

```diff
@@ -159,8 +158,20 @@ public IApiTypeRoot findTypeRoot(String qualifiedName) throws CoreException {
 	public String[] getPackageNames() throws CoreException {
 		if (fPackageNames == null) {
 			SortedSet<String> names = new TreeSet<>();
-			if (fPackageFragmentRoot.exists()) {
-				collectPackageNames(names, fPackageFragmentRoot);
+			// Try to use JDT API if we have a package fragment root and it corresponds to our container
+			boolean useJdtApi = false;
+			if (fPackageFragmentRoot != null && fPackageFragmentRoot.exists()) {
+				IResource rootResource = fPackageFragmentRoot.getResource();
+				// Use JDT API only if the package fragment root's resource matches our container
+				// This ensures we're querying the right location (e.g., output folder vs source folder)
+				if (rootResource != null && rootResource.equals(fRoot)) {
+					useJdtApi = true;
+					collectPackageNamesFromJdt(names, fPackageFragmentRoot);
+				}
+			}
+			// Fall back to filesystem scanning if JDT API is not applicable
+			if (!useJdtApi) {
+				collectPackageNamesFromFilesystem(names, fRoot);
```

Comment:
Why is this needed? Even with shared output folder JDT should be able to find all items ... JDT can even read from jar files and the classpath of a project also contains infos about output folder.

The actual output folder should never matter and discovering things from the source would even be better becuase we do not need any compiled classes fisrt.